op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -421,37 +454,57 @@ func runStateMigration(newDBPath string, opts stateMigrationOptions) error {
 	}
 	config.SetDeployments(deployments)
 
-	// Get latest block information from L1
 	var l1StartBlock *types.Block
 	client, err := ethclient.Dial(opts.l1RPC)
 	if err != nil {
 		return fmt.Errorf("cannot dial %s: %w", opts.l1RPC, err)
 	}
-
+	// If the L1 starting block tag is not set, we determine it dynamically by
+	// finding the most recent final L1 block at the time of the L2 fork block.
 	if config.L1StartingBlockTag == nil {
-		l1StartBlock, err = client.BlockByNumber(context.Background(), nil)
+		// Find the L1 starting block, the L2 fork block occurs 1 minute after the last celo L1 block.
+		opts.migrationBlockTime = celoL1Head.Time + 60
+		// Wait for the L2 start time before we calculate the L1 starting block.
+		tillL2Start := int64(opts.migrationBlockTime) - time.Now().Unix()
+		if tillL2Start >= 0 {
+			time.Sleep(time.Duration(tillL2Start+1) * time.Second)
```

Comment:
Why do we need to wait here? If the beacon chain API call below takes a few seconds anyway, why not add this check after that returns?